solutions/usaco-809.mdx

```diff
@@ -0,0 +1,67 @@
+---
+id: usaco-809
+source: USACO Bronze 2018 Febuary
+title: Taming the Herd
+author: Sathvik Chundru
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_taming_bronze_feb18.html)
+
+<LanguageSection>
+
+<JavaSection>
+	
+```java
+import java.util.*;
+import java.io.*;
+
+public class Taming{
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio("taming");
+		int n = io.nextInt();
+		int[] log = new int[100000];
+		for(int i = 0; i < n; i++){
+			log[i] = io.nextInt();
+		}
+		if(log[0] > 0){
+			io.println(-1);
+			io.close();
+			System.exit(0);
+		}
+		log[0] = 0;
+		int t = -1;
+		int req = 0;
+		int pos = 0;
+		for(int i = n - 1; i >= 0; i--){
+			if( t != -1 && log[i] != -1 && log[i] != t){
+				io.println(-1);
+				io.close();
+				System.exit(0);
+			}
+			if(t == -1){
```

Comment:
Can you add an explanation for this long pile of if statements? It's not super obvious at first glance what the purpose of all of these are.

Reply:
addressed this above